packages/core/lib/types/queueOptionsTypes.ts

```diff
@@ -5,11 +5,27 @@ import type { PayloadStoreConfig } from '../payload-store/payloadStoreTypes'
 import type { MessageHandlerConfig } from '../queues/HandlerContainer'
 import type { HandlerSpy, HandlerSpyParams } from '../queues/HandlerSpy'
 
-import type { Logger, TransactionObservabilityManager } from './MessageQueueTypes'
+import type {
+  Logger,
+  MessageProcessingResult,
+  TransactionObservabilityManager,
+} from './MessageQueueTypes'
 
 export type QueueDependencies = {
   errorReporter: ErrorReporter
   logger: Logger
+  messageMetricsManager?: MessageMetricsManager
+}
+
+export type ProcessedMessageMetadata = {
+  messageId: string
+  messageType: string
```

Comment:
wonder if we may also want to store a version

Reply:
Do you mean `schemaVersion`?

Reply:
yup!

Reply:
`schemaVersion` in metadata is specific to the message format that we use in `DomainEventEmitter`, and we may be interested in other parameters as well, so I added whole message as a property here and made it generic. Let me know if that looks good to you